### PR TITLE
Some more exhaust from the CHERIfication process

### PR DESCRIPTION
**Pull request by nwf**
Once again, pulling things that aren't strictly StrictProvenance-related out of #295 for stand-alone review.  No functional changes intended with this batch.

### CODE REVIEW

**Review by mjp41 — APPROVED** [review]
LGTM

**Comment by nwf** [comment]
Sorry, one more commit.

**Review by mjp41 — APPROVED** [review]

File: src/mem/largealloc.h [hunk 1]
```diff
@@ -290,13 +290,8 @@ namespace snmalloc
     LargeAlloc(MemoryProvider& mp) : memory_provider(mp) {}
 
     template<ZeroMem zero_mem = NoZero>
-    void* alloc(size_t large_class, size_t size)
+    void* alloc(size_t large_class, size_t rsize, size_t size)
     {
-      size_t rsize = bits::one_at_bit(SUPERSLAB_BITS) << large_class;
```
Comment (mjp41) [inline src/mem/largealloc.h:295]: Perhaps add an assertion to confirm 
```
(bits::one_at_bit(SUPERSLAB_BITS) << large_class ) == rsize
```

**Review by mjp41 — APPROVED** [review]

**Comment by nwf** [comment]
Sorry, yes, just one more again (c289637)

**Review by mjp41 — APPROVED** [review]

**Pull request merged**